src/libYARP_companion/src/Companion.cpp

```diff
@@ -1450,8 +1450,8 @@ int Companion::cmdDetectRos(bool write) {
     }
 
     std::string uri = NetworkBase::getEnvironment("ROS_MASTER_URI");
-    if (uri=="") {
-        fprintf(stderr, "ROS_MASTER_URI environment variable not set.\n");
+    if (uri=="" || uri.length() > 512) {
+        fprintf(stderr, "ROS_MASTER_URI environment variable not set or too long.\n");
```

Comment:
It would be nice as a user to know that too long means "more than 512 characters", so I can eventually take care of it with some trick if in my case ROS_MASTER_URI is longer.